File: java/src/tk/csparpa/gofdp/builder/Director.java
package tk.csparpa.gofdp.builder;

/**
 * 
 * This class is used by the clients to create new Product instances by using
 * an AbstractBuilder object.
 *
 */
public class Director {
	
	private AbstractBuilder builder;
	
	public Director(AbstractBuilder builder){
		this.builder = builder;
	}
	
	/**
	 * 
	 * This method builds the whole new Product instance. The creation
	 * of the single parts is delegated to the AbstractBuilder instance, as
	 * well as the provisioning of the instance
	 */
	public Product construct(){
		this.builder.buildPartA();
		this.builder.buildPartB();
		return this.builder.getProduct();
	}
}

File: python/builder/director.py
"""
This class is used by the clients to create new Product instances by using
an AbstractBuilder object.
"""
class Director():
    
    _builder = None
    
    def __init__(self, abstract_builder_subtype):
        self._builder = abstract_builder_subtype
    
    """ 
    This method builds the whole new Product instance. The creation
    of the single parts is delegated to the AbstractBuilder instance, as
    well as the provisioning of the instance
    """
    def construct(self):
        self._builder.build_part_A()
        self._builder.build_part_B()
        return self._builder.get_product()